citrination_client/client.py

```diff
@@ -191,43 +191,43 @@ def predict(self, model_name, candidates):
+        Upload a file, specifying source and dest paths a file (acts as the scp command)
+        :param source_path: The path to the file on the source host
+        :param dest_path: The path to the file where the contents of the upload will be written (on the dest host)
+        :return: The response object or a boolean False if the file was not uploaded
         """
-        if os.path.isdir(str(file_path)):
-            for path, subdirs, files in os.walk(file_path):
-                for name in files:
-                    if root_path:
-                       root = root_path
-                    else:
-                       root = str(file_path)
-                    success = self.upload_file(os.path.join(path, name), data_set_id, root)
-                    print(success)
-            message = {"message": "Upload of files in " + str(root) + " is complete."}
-            return json.dumps(message)
+        source_path = str(source_path)
+        if not dest_path:
+            dest_path = source_path
         else:
-            url = self._get_upload_url(data_set_id)
-            file_data = {"file_path": str(file_path)}
-            if root_path:
-                file_data["root_path"] = root_path
+            dest_path = str(dest_path)
 
+        if os.path.isdir(source_path):
+            for path, subdirs, files in os.walk(source_path):
+                for name in files:
+                    path_without_root_dir = path.split("/")[1:] + [name]
+                    current_dest_path = os.path.join(dest_path, *path_without_root_dir)
+                    current_source_path = os.path.join(path, name)
+                    success = self.upload(dataset_id, current_source_path, current_dest_path)
+            message = {"message": "Upload of files is complete."}
+            return json.dumps(message)
+        elif os.path.isfile(source_path):
+            url = self._get_upload_url(dataset_id)
+            file_data = { "file": { "dest_path": str(dest_path), "src_path": str(source_path)}}
             r = requests.post(url, data=json.dumps(file_data), headers=self.headers)
             if r.status_code == 200:
                 j = json.loads(r.content.decode('utf-8'))
                 s3url = self._get_s3_presigned_url(j)
-                with open(file_path, 'rb') as f:
+                with open(source_path, 'rb') as f:
                     r = requests.put(s3url, data=f)
                     if r.status_code == 200:
                         url_data = {'s3object': j['url']['path'], 's3bucket': j['bucket']}
                         requests.post(self._get_update_file_upload_url(j['file_id']),
                                       data=json.dumps(url_data), headers=self.headers)
                         message = {"message": "Upload is complete.",
-                                   "data_set_id": str(data_set_id),
+                                   "dataset_id": str(dataset_id),
```

Comment:
This is essentially an interface change in that the format of the return value has changed.  Is this not going to affect existing consumers?